### PR TITLE
`Vm enter scope` hint

### PR DESCRIPTION
**Pull request by toni-calvin**

### CODE REVIEW

File: pkg/types/exec_scopes.go [hunk 1]
```diff
@@ -14,6 +14,10 @@ func ExecutionScopesError(err error) error {
 	return errors.Wrapf(err, "Execution scopes error")
 }
 
+func (es *ExecutionScopes) Data() []map[string]interface{} {
+	return es.data
+}
+
```
Comment (fmoletta) [inline pkg/types/exec_scopes.go]: Remove

File: pkg/hints/memcpy_hints_test.go [hunk 2]
```diff
@@ -79,3 +79,39 @@ func TestExitScopeInvalid(t *testing.T) {
 	}
 
 }
+
+func TestEnterScope(t *testing.T) {
+	vm := NewVirtualMachine()
+	vm.Segments.AddSegment()
+	idsManager := SetupIdsForTest(
+		map[string][]*MaybeRelocatable{
+			"a": {NewMaybeRelocatableFelt(FeltFromUint64(17))},
```
Comment (fmoletta) [inline pkg/hints/memcpy_hints_test.go]: This is not used

File: pkg/types/exec_scopes.go [hunk 3]
```diff
@@ -44,7 +48,7 @@ func (es *ExecutionScopes) getLocalVariablesMut() (*map[string]interface{}, erro
 	if err != nil {
 		return nil, err
 	}
-	return &locals, err
+	return &locals, nil
```
Comment (fmoletta) [inline pkg/types/exec_scopes.go:47]: why

Comment (toni-calvin) [reply pkg/types/exec_scopes.go:47]: I think even if it's the same, its more readable to return a nil

**Review by fmoletta — APPROVED** [review]

**Review by entropidelic — APPROVED** [review]

**Pull request merged**